src/blockdata/transaction.rs

```diff
@@ -609,6 +609,24 @@ impl Transaction {
     pub fn is_explicitly_rbf(&self) -> bool {
         self.input.iter().any(|input| input.sequence < (0xffffffff - 1))
     }
+
+    /// Returns the nLockTime as an absolute timelock.
+    pub fn absolute_lock_time(&self) -> timelock::Abs {
+        timelock::Abs::from(self.lock_time)
+    }
+
+    /// Return the nSequence for `input_index` as a relative timelock.
+    pub fn relative_lock_time(&self, input_index: usize) -> timelock::Rel {
```

Comment:
I think this belongs rather on `TxIn`.

On `Transaction` you could implement something like `maximum_relative_lock_time` which provides the most strict `nSequence` that needs to be satisfied.

Reply:
There is no concept of 'maximum' for all inputs because some could be time based and some block based, right? 

(I moved `relative_lock_time` as suggested.)

Reply:
Ah, yes -- though you could have an accessor that returns the maximum height-based lock *and* maximum time-based lock. (It should probably be one accessor returning two values, rather than separate accessors, to remind the user that they need to check both normally.)

Reply:
Doe in, comit: `d6c19f9 Add maximum_relative_timelock method to Transaction`.

Not super clean, but shows the idea.